lib/perl/Genome/Model/ClinSeq/Command/UpdateAnalysis.pm

```diff
@@ -820,12 +839,42 @@ sub get_trsn{
 
   my %trsns;
   my $trsn_ref;
-
-  foreach my $instrument_data (@instrument_data){
-    my $trsn = $instrument_data->target_region_set_name;
-    if ($trsn){
-      $trsns{$trsn}=1;
-      $trsn_ref = $trsn;
+  
+  #When using validation-as-exome and my-trsn options, use user-defined target region set name (1=exome or 2=validation)
+  if ($self->validation_as_exome && $self->my_trsn) {
+    my $user_trsn = $self->my_trsn;
+    
+    #Check if a valid option for my-trsn and use or warn user and exit
+    if ($user_trsn == 1){
+      $user_trsn = 'exome';
+      $self->warning_message("Using '$user_trsn' target region specified by my-trsn.");
+    }elsif($user_trsn == 2){
+      $user_trsn = 'validation';
+      $self->warning_message("Using '$user_trsn' target region specified by my-trsn.");
+    }else{
+      die $self->error_message("Not a valid argument for my_trsn. 1='exome' or 2='validation'");
+    }
+    
+    #Assigns the target-region-set-name to the value defined by the user
```

Comment:
You might want to add a check after this foreach block in case none of the feature lists are of the correct content type which would result in `$trns_ref` not getting set at all.